src/control/cmd/daos_server/storage.go

```diff
@@ -24,6 +25,8 @@ import (
 	"github.com/daos-stack/daos/src/control/server/storage"
 )
 
+const iommuPath = "/sys/class/iommu"
```

Comment:
Seems like a good candidate to go into the lib/hardware stuff. We have a testable bit of code reaching into sysfs there. I'd suggest adding an interface (IOMMUDetector?) that we could mock for outside users, and have the sysfs.Provider implement an IsIOMMUEnabled method. That would replace iommuDetected().

Reply:
sounds good, I will put that in

Reply:
done